tools/Custom/JsonExtensions.cs

```diff
@@ -1,62 +1,85 @@
 {
     using Newtonsoft.Json.Linq;
+    using System;
     using System.Linq;
 
     public static class JsonExtensions
     {
         /// <summary>
-        /// Removes JSON properties that have a value of "defaultnull" and converts properties with values of "null" to actual JSON null values.
+        /// Recursively removes properties with the value "defaultnull" from a JSON structure
+        /// and replaces string values that are "null" with actual null values.
+        /// This method supports both JObject (JSON objects) and JArray (JSON arrays),
+        /// ensuring proper cleanup of nested structures.
         /// </summary>
-        /// <param name="jsonObject">The JObject to process and clean.</param>
-        /// <returns>
-        /// A JSON string representation of the cleaned JObject with "defaultnull" properties removed and "null" values converted to JSON null.
-        /// </returns>
+        /// <param name="token">The JToken (JObject or JArray) to process.</param>
+        /// <returns>The cleaned JSON string with "defaultnull" values removed and "null" strings converted to null.</returns>
         /// <example>
         /// JObject json = JObject.Parse(@"{""name"": ""John"", ""email"": ""defaultnull"", ""address"": ""null""}");
         /// string cleanedJson = json.RemoveDefaultNullProperties();
         /// Console.WriteLine(cleanedJson);
         /// // Output: { "name": "John", "address": null }
         /// </example>
-        public static string RemoveDefaultNullProperties(this JObject jsonObject)
+        public static string RemoveDefaultNullProperties(this JToken token)
         {
             try
             {
-                foreach (var property in jsonObject.Properties().ToList())
+                if (token is JObject jsonObject)
                 {
-                    if (property.Value.Type == JTokenType.Object)
+                    foreach (var property in jsonObject.Properties().ToList())
                     {
-                        RemoveDefaultNullProperties((JObject)property.Value);
-                    }
-                    else if (property.Value.Type == JTokenType.Array)
-                    {
-                        foreach (var item in property.Value)
+                        if (property.Value.Type == JTokenType.Object)
                         {
-                            if (item.Type == JTokenType.Object)
-                            {
-                                RemoveDefaultNullProperties((JObject)item);
-                            }
+                            RemoveDefaultNullProperties(property.Value);
+                        }
+                        else if (property.Value.Type == JTokenType.Array)
+                        {
+                            RemoveDefaultNullProperties(property.Value);
+                        }
+                        else if (property.Value.Type == JTokenType.String && property.Value.ToString() == "defaultnull")
```

Comment:
update the string comparisons to pass a comparison type. 
e.g
```suggestion
                        else if (property.Value.Type == JTokenType.String && property.Value.ToString().Equeals("defaultnull",StringComparison.Ordinal)
```